thunder/tests/distributed/test_ddp.py

```diff
@@ -448,42 +456,41 @@ def is_same_across_ranks(t):
-        if "te_linear" in bound_symbol.sym.name:
-            thunder_fp8_meta = bound_symbol._call_ctx[bound_symbol.sym.name].func.fp8_meta
-            te_fp8_meta = thunder_to_te_layer_map[bound_symbol.sym.name].fp8_meta
-            try:
-                # fwd tensor history
-                assert_close(thunder_fp8_meta["scaling_fwd"].scale, te_fp8_meta["scaling_fwd"].scale)
-                assert_close(thunder_fp8_meta["scaling_fwd"].scale_inv, te_fp8_meta["scaling_fwd"].scale_inv)
-                assert_close(thunder_fp8_meta["scaling_fwd"].amax_history, te_fp8_meta["scaling_fwd"].amax_history)
-                # bwd tensor history
-                assert_close(thunder_fp8_meta["scaling_bwd"].scale, te_fp8_meta["scaling_bwd"].scale)
-                assert_close(thunder_fp8_meta["scaling_bwd"].scale_inv, te_fp8_meta["scaling_bwd"].scale_inv)
-                assert_close(thunder_fp8_meta["scaling_bwd"].amax_history, te_fp8_meta["scaling_bwd"].amax_history)
-
-                # This has to be on all ranks so that the computation is not blocked
-                is_same_across_ranks(thunder_fp8_meta["scaling_fwd"].scale)
-                is_same_across_ranks(thunder_fp8_meta["scaling_fwd"].scale_inv)
-                # NOTE: TE forward tensor meta-data sync
-                # Syncing of FP8 meta-data happens in two step in the forward pass.
-                # 1. When we enter the fp8_autocast(), all the forward fp8 meta-data
-                # in global buffer is synced.
-                # See: https://github.com/NVIDIA/TransformerEngine/blob/6a9edc38bf9b941b7d369af5103fa8fe0b121d61/transformer_engine/pytorch/fp8.py#L409-L412
-                # 2. Post this, in the forward pass of the module in `prepare_forward`,
-                # we read from the global-buffer the synced meta-data.
-                # See: https://github.com/NVIDIA/TransformerEngine/blob/6a9edc38bf9b941b7d369af5103fa8fe0b121d61/transformer_engine/pytorch/module/base.py#L539-L545
-                # However, at the end of this forward pass, we have seen new inputs and outputs. Their amax are recorded on
-                # 0th row of `amax_history` (which will be synced only in the next forward pass).
-                # So, here we check that every row except for `0` is same.
-                is_same_across_ranks(thunder_fp8_meta["scaling_fwd"].amax_history[1:])
-                is_same_across_ranks(thunder_fp8_meta["scaling_bwd"].scale)
-                is_same_across_ranks(thunder_fp8_meta["scaling_bwd"].scale_inv)
-                is_same_across_ranks(thunder_fp8_meta["scaling_bwd"].amax_history)
-            except Exception as e:
-                # Return exceptions only for rank==0
-                if rank == 0:
-                    comparison_exceptions.append(e)
+    if not isinstance(
+        get_default_fp8_recipe(), MXFP8BlockScaling
+    ):  # MXFP8BlockScaling recipe doesn't have state like scale, amax_history.
```

Comment:
Where is the block scaling state stored? On tensor objects?

Reply:
For BlockScaling recipe, the scales are stored on the quantized tensor objects. Also, worth noting that unlike DelayedScaling, there is no need to preserve state across the iteration as scaling factors are computed dynamically.

https://github.com/NVIDIA/TransformerEngine/blob/884248f5788ad1d5561a55878cd772866efc46a8/transformer_engine/pytorch/tensor/_internal/mxfp8_tensor_base.py#L46-L64

https://github.com/NVIDIA/TransformerEngine/blob/884248f5788ad1d5561a55878cd772866efc46a8/transformer_engine/pytorch/tensor/mxfp8_tensor.py#L129-L153